src/util/util.h

```diff
@@ -201,6 +202,25 @@ void gridOperation(const QRectF& extent, double horz_spacing, double vert_spacin
 	hatchingOperation(extent, vert_spacing, vert_offset, rotation - M_PI / 2, process_line);
 }
 
+/**
+ * Tests whether three points form what we would call a corner. The function
+ * returns true when point2 lies at least quantum_size from continuation of
+ * line from point1 to anchor_point. quantum_size parameter is essential for
+ * use of this function in GUI code as for various zoom levels point1 position
+ * get quantized with varying step size.
+ *
+ * @param anchor_point Point that is being tested as corner candidate.
+ * @param point1 First anchor_point neighbor.
+ * @param point2 Second anchor_point neighbor.
+ * @param quantum_size How far can point2 lie from the line defined by
+ *        point1--anchor to be considered unaligned with the two other points.
+ * @return True if point2 is further than quantum_size from the line defined 
+ *         by point1--anchor_point or point2 is on the same side of 
+ *         anchor_point as point1.
+ */
+bool pointsFormCorner(const MapCoord& anchor_point, const MapCoord& point1,
+                      const MapCoord& point2, const qreal quantum_size);
```

Comment:
- There is the `const qreal` in the declaration again ([readability-avoid-const-params-in-decls](https://releases.llvm.org/8.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-avoid-const-params-in-decls.html)).
- I would appreciate a short unit test for this function at least, in `util_t.cpp`.
- I'm concerned that the MapCoord parameter order is not so obvious when looking at a calling line. How about this signature:
~~~
bool pointsFormCorner(const MapCoord& first_point, const MapCoord& inner_point,
                      const MapCoord& last_point, const qreal quantum_size);
~~~
where `inner_point` takes the role of the former `anchor_point`?

Reply:
Thanks, Kai, for the quick review. I can see you started using clang-tidy. I looked into making the clang helpers work with openSUSE yesterday but it looks like there is something broken in the tidy plugin handling. Will try harder to make it work.
I'm all fine with `const` removal from the declaration. I'll watch for this error next time. Regarding the parameter ordering for `Util::pointsFormCorner()`, the rename brought a very slight shift in the meaning which escaped my attention. With the current function name it makes sense to order them as "first, middle, last". The original meaning was "(test that a point) forms a corner" and the order was "point to be tested, the other two points".
Ad the test for `Util::pointsFormCorner()`, I'll add a patch to the series.

Reply:
I was regularly using iwyu and clang-tidy, for the files white listed in code-check-wrapper. But unfortunately I didn't realize that the '.clang-tidy' was ignored due to wrong syntax.

Reply:
So let's make a move on this one. I've incorporated the drop of `const` on function prototype and parameter reordering for `Util::pointsFormCorner()`.
I've also taken another look at `code-check-wrapper.sh` and maybe I've guessed how to use it. In any case, I'd be happy to see instructions for use in developer wiki if the check output is important part of review.